glean-core/src/debug.rs

```diff
@@ -112,10 +132,10 @@ where
     ///
     /// Validates the value in case a validation function is available.
     pub fn set(&mut self, value: T) -> bool {
-        let validated = self.validate(value);
-        if validated.is_some() {
+        let validated = self.validate(value.clone());
```

Comment:
Does passing by reference remove the need for the `clone`?  Since `validate` doesn't hold on to the value, it should at least theoretically be possible to avoid the memory copy.